Tasks/google-play-release/GooglePlay.js

```diff
@@ -89,30 +89,18 @@ currentEdit = currentEdit.then(function (res) {
     return updateTrack(packageName, track, apkVersionCodes, userFraction);
 });
 
-if (shouldAttachMetadata) {
-    var metadataRootPath = tl.getInput("metadataRootPath", true);
-    currentEdit = currentEdit.then(function (res) {
-        console.log(`Attempting to attach metadata to release...`);
-        return addMetadata(metadataRootPath);
-    });
-}
+currentEdit = currentEdit.then(function (res) {
+    tl.debug("Update metadata if specified...");
+    return attachMetadata(shouldAttachMetadata);
+})
 
-// This block will likely be deprecated by the metadata awareness
-try {
-    var stats = fs.statSync(changelogFile);
-    if (stats && stats.isFile()) {
-        apkVersionCodes.forEach(function(apkVersionCode) {
```

Comment:
This line is the root cause, the apkVersionCodes array is iterated before it's populated, because the promise chain has not yet started. The suggested fix looks good.